src/record/api.rs

```diff
@@ -410,102 +392,129 @@ fn convert_timestamp_to_string(value: u64) -> String {
 
 #[cfg(test)]
 mod tests {
+  use std::rc::Rc;
+
   use super::*;
   use chrono;
+  use schema::types::{ColumnDescriptor, ColumnPath, PrimitiveTypeBuilder};
+
+  /// Creates test column descriptor based on provided type parameters.
+  macro_rules! make_column_descr {
```

Comment:
Why this requires a macro but not a normal function?

Reply:
When adding decimal, I plan to add the second arm that will have signature `$physical_type:expr, $logical_type:expr, $length:expr, $precision:expr, $scale:expr`, so I could use the same name but different parameters. With function I would have to either use length, precision, and scale as 0 for ones that don’t support it, or have different function name. So I chose a macro instead.

Reply:
Ah, I see. Poor man's function overloading. 😀

Reply:
Ahaha:) Yes, you are right, I just thought it would be easier that way.